### PR TITLE
[MQTT] Mosquitto 2.0 and Configuration #1071

### PR DESCRIPTION
**Pull request by atrovato**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- ~~[ ] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)~~
- ~~[ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)~~
- ~~[ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).~~
- ~~[ ] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).~~

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

In case of embedded usage, now force to use mosquitto v2 container, and check the configuration, if `listener 1883` line is missing, add it to mosquitto config file.

### CODE REVIEW

**Comment by atrovato** [comment]
@VonOx didn't test all cases for now:
 1. mosquitto container already present, with bad config
 2. mosquitto container already present, with manually fixed config
 3. mosquitto container new creation (from scratch)

I will this afternoon (depends on free time).

File: server/services/mqtt/lib/updateContainer.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    // Reinstall container with explicit version
+    const newContainer = await this.installContainer(false);
+    await this.gladys.system.restartContainer(newContainer.id);
+
+    await this.gladys.variable.setValue(
+      CONFIGURATION.MQTT_MOSQUITTO_VERSION,
+      DEFAULT.MOSQUITTO_VERSION,
+      this.serviceId,
+    );
+    configuration.mqttPort = DEFAULT.MOSQUITTO_VERSION;
```
Comment (Pierre-Gilles) [inline server/services/mqtt/lib/updateContainer.js]: why does the MQTT port is equal to the MOSQUITTO_VERSION? seems weird

**Comment by Pierre-Gilles** [comment]
Very nice fixes! thanks for looking into it guys :)

**Comment by atrovato** [comment]
Works fine with new install, but I have env troubles while try to enter in update mode.

**Comment by VonOx** [comment]
@atrovato I've builded your PR for testing

Something is broken 
image

I'm running the container with this command
```
docker run -d --log-opt max-size=10m --restart=always --privileged --network=host --name gladys -e NODE_ENV=production -e SERVER_PORT=80 -e TZ=Europe/Paris -e SQLITE_FILE_PATH=/var/lib/gladysassistant/gladys-production.db -v /var/run/docker.sock:/var/run/docker.sock -v /var/lib/gladysassistantdev:/var/lib/gladysassistant -v /dev:/dev vonox/gladys:mqtt
```
Nothing exotic

**Comment by atrovato** [comment]
Oops I forgot to tell I built a docker image `atrovato/gladys:mqtt-mosquitto-v2` -- clicking on action ;) 

The check on network is based on hard coded gladys image name `dockerImage: 'gladysassistant/gladys'`.
I'm not sure we can get "currently running container".

So the workaround is to keep another container with original image.

**Comment by VonOx** [comment]
Oh ! my bad

Solution to improve detection:

`head -1 /proc/self/cgroup | cut -d/ -f3`
From inside container this return id, with id you can inspect every options

image
image

**Comment by atrovato** [comment]
I'll try this, thanks.

**Comment by Pierre-Gilles** [comment]
>head -1 /proc/self/cgroup | cut -d/ -f3 

Make sure that this is only executed when inside Docker. When developing on a Windows/Mac machine, this will not make good stuff ^^

**Comment by atrovato** [comment]
@Pierre-Gilles yes of course, check docker mode first.
@VonOx 3 modes tested, looks fine to me.

**Comment by atrovato** [comment]
Image (all arch): atrovato/gladys:mqtt-mosquitto-v2

**Comment by Pierre-Gilles** [comment]
@atrovato Awesome! I'll try this.
@VonOx Could you try this too and confirm it works? :)

**Comment by VonOx** [comment]
Alex, maybe I'm wrong but my mosquitto config is not updated.

To test I have made a fresh install , mosquitto container created ( with listener option )  => ok
So I stop both container , removed listener option and restarted gladys container => fail

config is not updated , gladys logs sayed connected , I dont understand.
```
2021-02-21T18:40:03+0100 <info> index.js:16 (Object.start) Starting MQTT service
2021-02-21T18:40:03+0100 <info> updateContainer.js:13 (MqttHandler.updateContainer) MQTT: checking for required changes...
2021-02-21T18:40:03+0100 <info> connect.js:38 (MqttClient.<anonymous>) Connected to MQTT server mqtt://localhost
2021-02-21T18:40:03+0100 <info> subscribe.js:12 (MqttHandler.subscribe) Subscribing to MQTT topic gladys/master/#
2021-02-21T18:40:03+0100 <info> index.js:15 (Object.start) Starting Bluetooth service
2021-02-21T18:40:04+0100 <info> bluetooth.initPresenceScanner.js:18 (BluetoothManager.initPresenceScanner) Bluetooth configuration: starting presence scanner
2021-02-21T18:40:04+0100 <info> index.js:16 (Object.start) Starting zwave service
2021-02-21T18:40:04+0100 <info> service.start.js:40 (Service.start) Service zwave is not configured, so it was not started.
2021-02-21T18:40:04+0100 <info> index.js:19 (Object.start) Starting Telegram service
2021-02-21T18:40:04+0100 <info> service.start.js:40 (Service.start) Service telegram is not configured, so it was not started.
2021-02-21T18:40:04+0100 <info> index.js:51 (Object.start) Starting CalDAV service
noble warning: adapter does not support Bluetooth Low Energy (BLE, Bluetooth Smart).
               Try to run with environment variable:
               [sudo] NOBLE_HCI_DEVICE_ID=x node ...
2021-02-21T18:40:04+0100 <info> index.js:18 (Object.start) Starting TP-Link service
2021-02-21T18:40:04+0100 <info> index.js:17 (Object.start) Starting eWeLink service
2021-02-21T18:40:04+0100 <info> service.start.js:40 (Service.start) Service ewelink is not configured, so it was not started.
2021-02-21T18:40:04+0100 <info> index.js:16 (Object.start) Starting RTSP service
2021-02-21T18:40:04+0100 <info> index.js:15 (Object.start) Starting Tasmota service
2021-02-21T18:40:04+0100 <info> subscribe.js:12 (MqttHandler.subscribe) Subscribing to MQTT topic stat/+/+
2021-02-21T18:40:04+0100 <info> subscribe.js:12 (MqttHandler.subscribe) Subscribing to MQTT topic tele/+/+
2021-02-21T18:40:04+0100 <info> index.js:18 (Object.start) Starting Philips Hue service
2021-02-21T18:40:04+0100 <info> index.js:14 (Object.start) Starting Xiaomi service
2021-02-21T18:40:04+0100 <info> index.js:13 (Object.start) Starting usb service
2021-02-21T18:40:04+0100 <info> index.js:63 (Server.<anonymous>) Server listening on port 80
```
image

--------------------------

Oh and i have discovered that gladys mqtt integration doen't start container if it's stopped. ( Trying to connect to a stopped container will always fail :) )

**Comment by atrovato** [comment]
> So I stop both container , removed listener option and restarted gladys container => fail

Once Gladys already applied migration (information stored in database), it doesn't apply it again.
I think you try to test the "update case". You have to configure Gladys to use embedded broker with an official image, it should install mosquitto:latest. Then try with my image, and you should get mosquitto:2 container and listener added to config.

> gladys mqtt integration doen't start container if it's stopped

I'm not sure Gladys should try to handle any potential human modification.
Watchtower should do this.

**Comment by Pierre-Gilles** [comment]
I just noticed something with the MQTT integration, the `/var/lib/gladysassistant` volume of the mosquitto container is hardcoded and cannot be changed! :o 

I was trying to make this work on another folder not to destroy my current installation, and launched Gladys like that: 


```
docker run -d \
--log-opt max-size=10m \
--restart=always \
--privileged \
--network=host \
--name gladys-2 \
-e NODE_ENV=production \
-e SERVER_PORT=8000 \
-e TZ=Europe/Paris \
-e SQLITE_FILE_PATH=/var/lib/gladysassistant/gladys-production.db \
-v /var/run/docker.sock:/var/run/docker.sock \
-v /root/gladys-test-mosquitto:/var/lib/gladysassistant \
atrovato/gladys:mqtt-mosquitto-v2
```

And the mosquitto container was started with a `/var/lib/gladysassistant` volume, thus destroying my other Gladys installation ^^ 

Maybe not in this release, but it could be great to change this behavior, and use the same folder as the SQLITE_FILE_PATH :)

**Comment by Pierre-Gilles** [comment]
@atrovato I tested this PR end-to-end on x64 server, and I confirm this work as expected ! :) Looks good to me !

**Review by Pierre-Gilles — APPROVED** [review]

**Pull request merged**